game/graphics/renderer.cpp

```diff
@@ -626,6 +711,56 @@ void Renderer::drawFxaa(Encoder<CommandBuffer>& cmd) {
   cmd.draw(Resources::fsqVbo());
   }
 
+void Renderer::applyCmaa2(Tempest::Encoder<Tempest::CommandBuffer>& cmd)
+{
+  uint32_t processCandidatesSetupFlag = 1;
+
+  static bool isFirstRun = true;
+  // initialization that is needed only on the first run. Make it run only in the first run
+  if (isFirstRun) {
+    cmd.setUniforms(*cmaa2.prepareDispatchIndirectArguments, cmaa2.prepareDispatchIndirectArgumentsUbo, &processCandidatesSetupFlag, sizeof(uint32_t));
+    cmd.dispatch(1);
+    isFirstRun = false;
+    }
+
+  // detect edges
+  {
+    const IVec3 inputGroupSize = cmaa2.detectEdges2x2->workGroupSize();
+    const IVec3 outputGroupSize = inputGroupSize - IVec3(2, 2, 0);
+
+    uint32_t groupCountX = static_cast<uint32_t>((cmaa2.sceneTonemappedUav.size().w + outputGroupSize.x * 2 - 1) / (outputGroupSize.x * 2));
```

Comment:
In c++ no need to use `static_cast` - it can be replaced with constructor-cast:
`static_cast<uint32_t>(epr) -> uint32_t(expr)`